docs/Contributing/dev-env.rst

```diff
@@ -0,0 +1,94 @@
+Set up a Development Environment
+********************************
+
+Contributing code to the Freeciv21 project or contributing to any of the games that the Longturn.net community
+manages requires a bit of setup. This document should get you up and running with the basics. It should be
+noted, that you typically only have to go through this process once unless you setup a new workstation.
+
+
+Workstation
+===========
+
+Freeciv21 can be developed on Linux, Windows and MacOS. Any current version of these OS's is acceptible. For
+Windows, you will need to setup the Msys2 environment to do development. Refer to :doc:`msys2` for more
+information. :strong:`Linux` is the preferred platform.
+
+Technically all you need is a text editor of some kind to edit the files, but most people prefer to use an
+IDE.
+
+All platforms can use `KDevelop <https://www.kdevelop.org/download>`_. However, there are some caveats:
+
+* On Windows, due to the nature of the integration with Msys2, native compilation and debugging is not
+  supported.
+* On MacOS, KDevelop is still considered experimental.
+
+For the best results, especially if you are editing game code and not just Longturn game rulesets or
+documentation, you will want :strong:`Linux` to be your worksation OS. Many of the current developers use a
+Debain variant such as Ubuntu. Instructions for getting all of the tools needed for Debian Linux can be found
+in :doc:`../General/install`. Refer to the section titled `Debian Linux Notes`. Don't follow the last few
```

Comment:
I am not sure if we need to talk about what the contributor's workflow is. I am pretty sure they already have that figured out before they start contributing.